main.py

```diff
@@ -1680,6 +1706,23 @@ def timer():
                         config["last_weighting_update"] = str(current_milli_time())
                         cur.execute("UPDATE config SET value = %s WHERE name = 'last_weighting_update'",
                                     [config["last_weighting_update"]])
+
+                    # pudding expiry?
+                    now = datetime.datetime.now()
+                    ymdNow = now.strftime("%Y-%m-%d")
+                    if ymdNow != config["last_pudding_check"]:
+                        config["last_pudding_check"] = ymdNow
+                        cur.execute("UPDATE config SET value = %s WHERE name = 'last_pudding_check'", [ymdNow])
+                        
+                        if now.day == 1:
+                            # move pudding down a category, alert people with expiring pudding
+                            cur.execute("UPDATE users SET puddingExpiring = puddingPrevious, puddingPrevious = puddingCurrent, puddingCurrent = 0")
+                            cur.execute("SELECT name, puddingExpiring FROM users WHERE puddingExpiring > 0")
+                            for userRow in cur.fetchall():
+                                self.message("#%s" % userRow[0], "You have %d pudding expiring on the 8th of this month. !pudding to see your balance and to spend it." % userRow[1], True)
```

Comment:
"8th of this month" – you avoid hardcoding the name of rarities, which are arguably much more fundamental, but then hardcode the expiry day of month?